### PR TITLE
warn toolchain on different platform

### PR DESCRIPTION
**Pull request by frbimo**
**Fixes** 
#2520 

**Testing**
`cargo test`

### CODE REVIEW

**Comment by frbimo** [comment]
@kinnison kindly reminder*

File: src/cli/rustup_mode.rs [hunk 1]
```diff
@@ -904,6 +906,32 @@ fn update(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
...
+                Ok(x) => x.has_triple(),
+                _ => false,
+            };
+
+            if toolchain_has_triple {
+                let host_arch = TargetTriple::from_host_or_build();
+                match ToolchainDesc::from_str(name) {
+                    Ok(toolchain_desc) => {
+                        let target_triple = toolchain_desc.target.clone();
+                        if host_arch.ne(&target_triple) {
```
Comment (kinnison) [inline src/cli/rustup_mode.rs]: Why did you feel the need to clone that here?

**Review by kinnison — COMMENTED** [review]
There's only one tiny niggle around what I feel might be a wasted `clone()` otherwise this looks excellent.  I'm especially grateful you added a test for this.

File: src/cli/rustup_mode.rs (hunk 1 above)
Comment (frbimo) [reply src/cli/rustup_mode.rs]: fixed

**Comment by frbimo** [comment]
@kinnison ptal

**Review by kinnison — APPROVED** [review]
This looks good to me.  One final chance for @rbtcollins to have a look in case the approach or wording isn't to his taste, otherwise I'll look to merge it fairly soon.

**Comment by rbtcollins** [comment]
Should this ask interactively perhaps?

**Comment by kinnison** [comment]
I'm loathe to add an interactive prompt to a part of `rustup` which has never been interactive before now.  If we think interactivity makes sense then we have to detect if there's a tty on stdin and stuff like that, otherwise it'd be a pain for anyone who has automated this in CI or similar.  Also we simply don't know for sure so warning non-interactively is probably enough.

**Comment by kinnison** [comment]
I rebased to clean up the conflict and am minded to merge this as I remain of the opinion that interactive prompting doesn't seem ideal to me.

**Comment by kinnison** [comment]
The CI badness is snapcraft being odd.

**Pull request merged**